src/services/recordingService.ts

```diff
@@ -137,19 +139,57 @@ export default class RecordingService {
     });
   };
 
-  private notifyRecordingTask = async (filePath: string, file_size: number) => {
+  private postProcess = async (filePath: string, file_size: number) => {
+    if (
+      !this.recorder.post_processing_scripts ||
+      !this.recorder.post_processing_scripts.length
+    ) {
+      return;
+    }
+    filePath = `${this.recorder.copy_to_path.main_path}/${filePath}`;
+
+    const data: PostProcessScriptData = {
+      recording_id: this.recordId,
+      room_table_id: Number(this.roomTableId),
+      room_sid: this.roomSid,
+      file_path: filePath, // this will be the full path of the file
+      file_size: file_size,
+      recorder_id: this.recorder.id,
+    };
+    const toSend = JSON.stringify(data);
+
+    for (let i = 0; i < this.recorder.post_processing_scripts.length; i++) {
+      const script = this.recorder.post_processing_scripts[i];
+      if (fs.existsSync(script)) {
+        if (typeof process.env.TS_NODE_DEV !== 'undefined') {
+          fork(script, [toSend], {
+            execArgv: ['-r', 'ts-node/register'],
+          });
+        } else {
+          fork(toSend, [toSend]);
+        }
+      }
+    }
+  };
```

Comment:
Maybe instead of fork we can use exec? This way we can have custom commands as scripts not just js files

Reply:
From `post_processing_scripts/example.js` you can use `exec` or anything.

Reply:
Ok then I think this will work I will try to create the opencast script, if I run into anything gonna let you know.